<javaelement>
	<package>
		java.sql
	</package>
	<type>
		<name>
			public class BatchUpdateException  extends SQLException
		</name>
		<description>
			The subclass of SQLException thrown when an error occurs during a batch update operation. In addition to the information provided by SQLException, a BatchUpdateException provides the update counts for all commands that were executed successfully during the batch update, that is, all commands that were executed before the error occurred. The order of elements in an array of update counts corresponds to the order in which commands were added to the batch. After a command in a batch update fails to execute properly and a BatchUpdateException is thrown, the driver may or may not continue to process the remaining commands in the batch. If the driver continues processing after a failure, the array returned by the method BatchUpdateException.getUpdateCounts will have an element for every command in the batch rather than only elements for the commands that executed successfully before the error. In the case where the driver continues processing commands, the array element for any command that failed is Statement.EXECUTE_FAILED.
		</description>
		<methods>
			<method>
				<name>
					public BatchUpdateException(String reason,                      String SQLState,                      int vendorCode,                      int[] updateCounts)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given reason, SQLState, vendorCode and updateCounts. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(String reason,                      String SQLState,                      int[] updateCounts)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given reason, SQLState and updateCounts. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method. The vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(String reason,                      int[] updateCounts)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given reason and updateCounts. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method. The SQLState is initialized to null and the vender code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(int[] updateCounts)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given updateCounts. initialized by a call to the Throwable.initCause(java.lang.Throwable) method. The reason and SQLState are initialized to null and the vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException()
				</name>
				<description>
					Constructs a BatchUpdateException object. The reason, SQLState and updateCounts are initialized to null and the vendor code is initialized to 0. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(Throwable cause)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given cause. The SQLState and updateCounts are initialized to null and the vendor code is initialized to 0. The reason is initialized to null if cause==null or to cause.toString() if cause!=null.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(int[] updateCounts,                      Throwable cause)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given cause and updateCounts. The SQLState is initialized to null and the vendor code is initialized to 0. The reason is initialized to null if cause==null or to cause.toString() if cause!=null.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(String reason,                      int[] updateCounts,                      Throwable cause)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given reason, cause and updateCounts. The SQLState is initialized to null and the vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(String reason,                      String SQLState,                      int[] updateCounts,                      Throwable cause)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given reason, SQLState,cause, and updateCounts. The vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public BatchUpdateException(String reason,                      String SQLState,                      int vendorCode,                      int[] updateCounts,                      Throwable cause)
				</name>
				<description>
					Constructs a BatchUpdateException object initialized with a given reason, SQLState, vendorCode cause and updateCounts.
				</description>
				
			</method>
			<method>
				<name>
					public int[] getUpdateCounts()
				</name>
				<description>
					Retrieves the update count for each update statement in the batch update that executed successfully before this exception occurred. A driver that implements batch updates may or may not continue to process the remaining commands in a batch when one of the commands fails to execute properly. If the driver continues processing commands, the array returned by this method will have as many elements as there are commands in the batch; otherwise, it will contain an update count for each command that executed successfully before the BatchUpdateException was thrown. The possible return values for this method were modified for the Java 2 SDK, Standard Edition, version 1.3. This was done to accommodate the new option of continuing to process commands in a batch update after a BatchUpdateException object has been thrown.
				</description>
				
			</method>
			
		</methods>
		
	</type>
	
</javaelement>
